### PR TITLE
CIRC-436: Implementation for request-preference-storage

### PR DESCRIPTION
**Pull request by andrii-paias**
## Purpose
Store user request preferences in mod-circulation-storage

## Approach
Add CRUD endpoints and implement them by using PgUtil class

### CODE REVIEW

File: ramls/request-preference.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "id": {
+      "description": "Unique request preference ID",
+      "type": "string"
+    },
+    "userId": {
+      "description": "UUID of user associated with this request preference",
+      "type": "string"
+    },
+    "holdShelf": {
+      "description": "Whether 'Hold shelf' option is available to the user.",
```
Comment (sduvvuri-ebsco) [inline ramls/request-preference.json]: Ah, I think I misunderstood this. My understanding was that its whether "holdShelf" or "delivery" option was selected by the user - not whether which of these options is available to user.

File: ramls/request-preference.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "metadata": {
+      "description": "Metadata about creation and changes to request preference",
+      "$ref": "raml-util/schemas/metadata.schema",
+      "readonly": true
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "userId",
+    "holdShelf",
```
Comment (sduvvuri-ebsco) [inline ramls/request-preference.json]: holdShelf is required by default, is delivery required as well?

Comment (dmtkachenko) [reply ramls/request-preference.json]: yes, it has to defined also to understand whether delivery = true or false

Comment (andrii-paias) [reply ramls/request-preference.json]: holdShelf is supposed to be always be true (i.e. UI checkbox is always checked), delivery can be false, but I think it still makes sense to always have this field in request

**Pull request merged**